### PR TITLE
bpf/test: Get rid of 4.9 leftovers

### PR DESCRIPTION
**Pull request by brb**
Misc cleanup which is part of \[1\].

\[1\]: https://github.com/cilium/cilium/issues/22116

NOTE: merge it only if we agree on https://github.com/cilium/cilium/pull/23381#issuecomment-1406239498

### CODE REVIEW

**Review by nebril — APPROVED** [review]

File: bpf/Makefile [hunk 1]
```diff
@@ -86,8 +84,8 @@ MAX_BASE_OPTIONS = -DSKIP_DEBUG=1 -DENABLE_IPV4=1 -DENABLE_IPV6=1 \
 	-DPOLICY_VERDICT_NOTIFY=1 -DALLOW_ICMP_FRAG_NEEDED=1 -DENABLE_IDENTITY_MARK=1 \
 	-DMONITOR_AGGREGATION=3 -DCT_REPORT_FLAGS=0x0002 -DENABLE_HOST_FIREWALL=1 \
 	-DENABLE_ICMP_RULE=1 -DENABLE_CUSTOM_CALLS=1 -DENABLE_SRV6=1 -DENABLE_L7_LB=1
-ifeq ("$(KERNEL)","49")
-# IPSec is incompatible with BPF NodePort so we only enable on 4.9.
+ifeq ("$(KERNEL)","419")
+# IPSec is incompatible with BPF NodePort so we only enable on 4.19.
 MAX_BASE_OPTIONS += -DENABLE_IPSEC=1 -DIP_POOLS=1
```
Comment (pchaigno) [inline bpf/Makefile]: Let's just drop this special case. The Makefile is just about generating the largest BPF programs possible and KPR is clearly impacting complexity more than IPsec.

Longer term we probably want to cover IPsec in complexity tests themselves (i.e., `bpf/complexity-tests/`) but that's orthogonal to this PR.

**Review by pchaigno — APPROVED** [review]

**Comment by ti-mo** [comment]
The tip of the iceberg. :sweat_smile: Backports necessary, though?

**Comment by brb** [comment]
> Backports necessary, though?

Nice to have, but not critical.

**Comment by julianwiedmann** [comment]
> > Backports necessary, though?
> 
> Nice to have, but not critical.

Let's drop the 1.13 backport label, we're past the "nice to have" stage :).

**Pull request merged**